cmd/scale.go

```diff
@@ -237,8 +239,31 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 				vmsToDelete = append(vmsToDelete, indexToVM[i])
 			}
 
-			if orchestratorInfo.OrchestratorType == api.Kubernetes {
-				err = sc.drainNodes(vmsToDelete)
+			switch orchestratorInfo.OrchestratorType {
+			case api.Kubernetes:
+				kubeConfig, err := acsengine.GenerateKubeConfig(sc.containerService.Properties, sc.location)
+				if err != nil {
+					log.Errorf("failed to generate kube config: %v", err)
+					return err
+				}
+				err = sc.drainNodes(kubeConfig, vmsToDelete)
+				if err != nil {
+					log.Errorf("Got error %+v, while draining the nodes to be deleted", err)
+					return err
+				}
+			case api.OpenShift:
+				bundle := bytes.NewReader(sc.containerService.Properties.OrchestratorProfile.OpenShiftConfig.ConfigBundles["master"])
+				fs, err := filesystem.NewTGZReader(bundle)
+				if err != nil {
+					log.Errorf("failed to read master bundle: %v", err)
+					return err
+				}
+				kubeConfig, err := fs.ReadFile("etc/origin/master/admin.kubeconfig")
+				if err != nil {
+					log.Errorf("failed to read kube config: %v", err)
+					return err
+				}
+				err = sc.drainNodes(string(kubeConfig), vmsToDelete)
```

Comment:
the drain node calls can be removed from the case statement, the only difference to account for is getting the kube config to pass in.

Reply:
however the drain node calls still must only run when the orchestrator is k8s or ocp - that's why they're in the switch still